### PR TITLE
Prevent panic when `JSON_CONSOLE` is enabled

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

File: core/logger/zap.go [hunk 1]
```diff
@@ -52,7 +57,7 @@ func newZapLogger(cfg ZapLoggerConfig) (Logger, func() error, error) {
 		config:            cfg,
 		pollDiskSpaceStop: make(chan struct{}),
 		pollDiskSpaceDone: make(chan struct{}),
-		SugaredLogger:     zap.New(core).Sugar(),
+		SugaredLogger:     zap.New(core, []zap.Option{zap.ErrorOutput(errWriter)}...).Sugar(),
```
Comment (jmank88) [inline core/logger/zap.go:60]:
```suggestion
		SugaredLogger:     zap.New(core, zap.ErrorOutput(errWriter)).Sugar(),
```

File: core/logger/zap.go [hunk 2]
```diff
@@ -168,14 +183,17 @@ func (l *zapLogger) Named(name string) Logger {
...
+	}
 	cores := []zapcore.Core{
 		// The console core is what we want to be unique per root, so we spin a new one here
-		newConsoleCore(newLogger.config),
+		consoleCore,
 	}
 	extraCores, err := newCores(newLogger.config)
 	cores = append(cores, extraCores...)
-	core := zap.New(zapcore.NewTee(cores...)).WithOptions(zap.AddCallerSkip(l.callerSkip))
+	core := zap.New(zapcore.NewTee(cores...), []zap.Option{zap.ErrorOutput(errWriter)}...).WithOptions(zap.AddCallerSkip(l.callerSkip))
```
Comment (jmank88) [inline core/logger/zap.go:196]:
```suggestion
	core := zap.New(zapcore.NewTee(cores...), zap.ErrorOutput(errWriter)).WithOptions(zap.AddCallerSkip(l.callerSkip))
```

File: core/logger/zap.go [hunk 3]
```diff
@@ -168,14 +171,17 @@ func (l *zapLogger) Named(name string) Logger {
...
+	}
 	cores := []zapcore.Core{
 		// The console core is what we want to be unique per root, so we spin a new one here
-		newConsoleCore(newLogger.config),
+		consoleCore,
 	}
 	extraCores, err := newCores(newLogger.config)
 	cores = append(cores, extraCores...)
-	core := zap.New(zapcore.NewTee(cores...)).WithOptions(zap.AddCallerSkip(l.callerSkip))
+	core := zap.New(zapcore.NewTee(cores...), zap.ErrorOutput(errWriter)).WithOptions(zap.AddCallerSkip(l.callerSkip))
```
Comment (jmank88) [inline core/logger/zap.go:184]: nit optimization
```suggestion
	core := zap.New(zapcore.NewTee(cores...), zap.ErrorOutput(errWriter), zap.AddCallerSkip(l.callerSkip))
```

File: core/logger/zap.go [hunk 4]
```diff
@@ -86,31 +90,30 @@ func newCores(cfg ZapLoggerConfig) ([]zapcore.Core, error) {
...
-		)
-	}
-
 	return cores, err
 }
 
-func newConsoleCore(cfg ZapLoggerConfig) zapcore.Core {
-	filteredLogLevels := zap.LevelEnablerFunc(cfg.Level.Enabled)
-
+func newConsoleCore(cfg ZapLoggerConfig) (zapcore.Core, zapcore.WriteSyncer, error) {
```
Comment (connorwstein) [inline core/logger/zap.go:96]: this is no longer just console - could be anything in paths, so maybe rename to just newCore ?

File: core/logger/zap.go [hunk 5]
```diff
@@ -168,14 +171,17 @@ func (l *zapLogger) Named(name string) Logger {
...
+	consoleCore, errWriter, err := newConsoleCore(newLogger.config)
+	if err != nil {
+		return nil, err
+	}
 	cores := []zapcore.Core{
 		// The console core is what we want to be unique per root, so we spin a new one here
-		newConsoleCore(newLogger.config),
+		consoleCore,
 	}
 	extraCores, err := newCores(newLogger.config)
```
Comment (connorwstein) [inline core/logger/zap.go:182]: should rename this as well - to just newDiskCore

Comment (vyzaldysanchez) [reply core/logger/zap.go:182]: this might as well get removed, since we already have a `newdiskcore`, and perform the validation where this is used

**Review by jmank88 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Pull request merged**